feature-libs/asm/components/customer-list/customer-list.component.ts

```diff
@@ -0,0 +1,243 @@
+import { Component, OnInit } from '@angular/core';
+import { AsmConfig, AsmService } from '@spartacus/asm/core';
+import {
+  CustomerListColumnActionType,
+  CustomerListsPage,
+  CustomerSearchOptions,
+  CustomerSearchPage,
+} from '@spartacus/asm/root';
+import { SortModel, TranslationService, User } from '@spartacus/core';
+import {
+  BREAKPOINT,
+  BreakpointService,
+  FocusConfig,
+  ICON_TYPE,
+  ModalService,
+} from '@spartacus/storefront';
+import { combineLatest, Observable, of } from 'rxjs';
+import { filter, map, switchMap, tap } from 'rxjs/operators';
+import { CustomerListActionEvent } from './customer-list.model';
+
+@Component({
+  selector: 'cx-customer-list',
+  templateUrl: './customer-list.component.html',
+})
+export class CustomerListComponent implements OnInit {
+  private PAGE_SIZE = 5;
+
+  focusConfig: FocusConfig = {
+    trap: true,
+    block: true,
+    autofocus: 'customer-list-selector',
+    focusOnEscape: true,
+  };
+
+  iconTypes = ICON_TYPE;
+
+  BREAKPOINT = BREAKPOINT;
+
+  selectedUserGroupId: string | undefined;
+
+  customerSearchPage$: Observable<CustomerSearchPage | null>;
+
+  customerListsPage$: Observable<CustomerListsPage | undefined>;
+
+  selectedCustomer: User;
+
+  currentPage = 0;
+
+  maxPage = 0;
+
+  loaded = false;
+
+  sorts: SortModel[] | null;
+
+  sortCode: string | undefined;
+
+  breakpoint$: Observable<BREAKPOINT>;
+
+  customerListConfig = this.asmConfig?.asm?.customerList;
+
+  constructor(
+    protected modalService: ModalService,
+    protected asmService: AsmService,
+    protected breakpointService: BreakpointService,
+    protected asmConfig: AsmConfig,
+    protected translation: TranslationService
+  ) {
+    this.breakpoint$ = this.getBreakpoint();
+  }
+
+  ngOnInit(): void {
+    this.customerListsPage$ = this.asmService
+      .getCustomerLists()
+      .pipe(
+        filter((queryState) => queryState.loading === false),
```

Comment:
can we move this `filter` and `map` into service? So, calling `this.asmService.getCustomerLists()` just get observable of customerListsPage.